File: backend/service/src/test/java/club/smileboy/app/mlnlco/service/lambda/serializableTests.java
package club.smileboy.app.mlnlco.service.lambda;

import club.smileboy.app.mlnlco.model.entity.UserEntity;
import com.baomidou.mybatisplus.core.toolkit.support.SFunction;
import com.baomidou.mybatisplus.core.toolkit.support.SerializedLambda;
import org.junit.jupiter.api.Test;

import java.util.Date;

/**
 * @author JASONJ
 * @date 2022/7/9
 * @time 12:00
 * @description mybatis plus lambda测试
 **/
public class serializableTests {
    @Test
    public void test(){
        final SFunction<UserEntity, Date> getCreateTime = UserEntity::getCreateTime;
        System.out.println(SerializedLambda.extract(getCreateTime));
    }
}

File: backend/service/src/test/kotlin/club/smileboy/app/mlnlco/service/serializable/SerializableTests.kt
package club.smileboy.app.mlnlco.service.lambda.serializable

import club.smileboy.app.mlnlco.model.entity.UserEntity
import com.baomidou.mybatisplus.core.toolkit.support.SFunction
import com.baomidou.mybatisplus.core.toolkit.support.SerializedLambda
import org.junit.jupiter.api.Test

/**
 * @author JASONJ
 * @date 2022/7/9
 * @time 11:37
 * @description 序列化测试
 **/
class SerializableTests {
    @Test
    fun test() {

        // error
//        println(SerializedLambda.extract(SFunction<Any,Any> { UserEntity::getCreateTime }).implMethodName)
    }
}